writeengine/wrapper/writeengine.cpp

```diff
@@ -1690,34 +1929,70 @@ int WriteEngineWrapper::insertColumnRecs(const TxnID& txnid,
+            int firstHalfCount = totalRow - rowsLeft;
             for (unsigned i = 0; i < colStructList.size(); i++)
             {
                 colOp = m_colOp[op(colStructList[i].fCompressionType)];
                 width = colStructList[i].colWidth;
-                successFlag = colOp->calculateRowId(lastRid, BYTE_PER_BLOCK / width, width, curFbo, curBio);
+                if (firstHalfCount)
+		{
+                    BRM::CPInfo* cpInfoP = getCPInfoToUpdateForUpdatableType(colStructList[i], &maxMins[i].fSplitMaxMinInfo[0]);
+                    RID thisRid = rowsLeft ? lastRid : lastRidNew;
+                    successFlag = colOp->calculateRowId(thisRid, BYTE_PER_BLOCK / width, width, curFbo, curBio);
 
-                if (successFlag)
-                {
-                    if (curFbo != lastFbo)
+                    if (successFlag)
                     {
-                        RETURN_ON_ERROR(AddLBIDtoList(txnid,
-                                                      colStructList[i],
-                                                      curFbo,
-						      dummyCPInfo));
+                        if (curFbo != lastFbo)
+                        {
+                            RETURN_ON_ERROR(AddLBIDtoList(txnid,
+                                                          colStructList[i],
+                                                          curFbo, cpInfoP));
+                        }
                     }
-                }
+                    maxMins[i].fSplitMaxMinInfoPtrs[0] = cpInfoP;
+		}
+                if (rowsLeft)
+		{
+                    BRM::CPInfo* cpInfoP = getCPInfoToUpdateForUpdatableType(colStructList[i], &maxMins[i].fSplitMaxMinInfo[1]);
+		    if (cpInfoP)
+		    {
+                        RETURN_ON_ERROR(GetLBIDRange(newExtentsStartingLbids[i], colStructList[i], *cpInfoP));
+		    }
+
+                    maxMins[i].fSplitMaxMinInfoPtrs[1] = cpInfoP;
+		}
             }
         }
 
-        markTxnExtentsAsInvalid(txnid);
+        markTxnExtentsAsInvalid(txnid); // moved here as it can be source of bug.
 
         //----------------------------------------------------------------------
         // Write row(s) to database file(s)
         //----------------------------------------------------------------------
-        rc = writeColumnRec(txnid, cscColTypeList, colStructList, colOldValueList, rowIdArray, newColStructList, colNewValueList, tableOid, useTmpSuffix); // @bug 5572 HDFS tmp file
+        std::vector<BRM::CPInfo> cpinfoList;
+        for (auto& splitCPInfo : maxMins)
+        {
+            for (i = 0; i < 2; i ++)
+            {
+                BRM::CPInfo* cpInfo = splitCPInfo.fSplitMaxMinInfoPtrs[i];
+                if (cpInfo)
+                {
+                    cpinfoList.push_back(*cpInfo);
+                    cpinfoList[cpinfoList.size() - 1].seqNum = -1;
+                }
+            }
+        }
+        rc = BRMWrapper::getInstance()->setExtentsMaxMin(cpinfoList);
```

Comment:
I didn't get why do we call setExtentsMaxMin() twice. Here for every column and later at 2030 after writeColumnRec and squeezeValidCPInfos. Could you ellaborate on this?